rust/tests/write_exploration.rs

```diff
@@ -175,12 +175,13 @@ impl DeltaWriter {
             let mut first = true;
 
             for k in partition_cols.iter() {
-                let partition_value =
-                    partition_values
-                        .get(k)
-                        .ok_or(DeltaWriterError::MissingPartitionColumn {
-                            col_name: k.to_string(),
-                        })?;
+                let partition_value = partition_values
+                    .get(k)
+                    .ok_or(DeltaWriterError::MissingPartitionColumn {
+                        col_name: k.to_string(),
+                    })?
+                    .clone()
+                    .unwrap_or("__HIVE_DEFAULT_PARTITION__".to_string());
```

Comment:
This reminded me that we are doing partition filtering wrong in `get_files_by_partitions`. We are parsing Action.path to  reconstruct the partition values, but instead, we should be using `Action.partition_values` or `Action.partition_values_parsed` directly.

I will file a separate issue to track this.

Reply:
cc @fvaleye https://github.com/delta-io/delta-rs/issues/358.